This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sun, 1 Oct 2023 14:14:37 +0200
Subject: [PATCH 1/1] Prevent pseudo modifiers from being defined

Add a definition for _NO_W32_PSEUDO_MODIFIERS so that windows.h will
not define the IN/OUT macro. Fixes a compilation error on MinGW-w64.

Upstream-Status: Submitted [https://github.com/openslide/openslide/pull/507]

diff --git a/src/openslide-private.h b/src/openslide-private.h
index 1111111..2222222 100644
--- a/src/openslide-private.h
+++ b/src/openslide-private.h
@@ -410,4 +410,9 @@ extern const int16_t _openslide_B_Cb[256];
 #define ftello _OPENSLIDE_POISON(_openslide_ftell_)
 #endif
 
+#ifdef _WIN32
+// Prevent windows.h from defining the IN/OUT macro
+#define _NO_W32_PSEUDO_MODIFIERS
+#endif
+
 #endif
